tests/unit/s2n_client_auth_handshake_test.c

```diff
@@ -467,10 +467,10 @@ int main(int argc, char **argv)
         size_t test_case_count = 0;
 
         for (size_t client_i = 0; client_i < s2n_array_len(all_options); client_i++) {
-            EXPECT_TRUE(test_case_count < s2n_array_len(test_cases));
             for (size_t server_i = 0; server_i < s2n_array_len(all_options); server_i++) {
                 for (size_t cert_i = 0; cert_i <= 1; cert_i++) {
                     for (size_t version = S2N_TLS12; version <= S2N_TLS13; version++) {
+                        EXPECT_TRUE(test_case_count < s2n_array_len(test_cases));
                         test_cases[test_case_count].client_auth_type = all_options[client_i];
```

Comment:
Instead of manually setting each member, could you use an initializer list (or whatever the proper C name is) and then add `-Wmissing-field-initializers`? This way the build would fail if someone is initializing a struct without setting a field.

Also enabling that warning only produces a single warning in our codebase 🥳. If you don't want to touch the build config in this PR, then let's at least open an issue for enabling that.

Reply:
This line is just a memory safety check, unrelated to the real problem.

The problem also isn't that some members aren't being initialized. They're all being set, they're just being overridden. The loop is just wrong. So wrong you may be missing how wrong it is :) Maybe this example will help: https://godbolt.org/z/GajM9Kxn6

If by "initializer list" you mean like test_cases = { blah blah blah }, then that's not ideal here because we need to cover all possible combos of the fields. Writing it out manually will be prone to mistakes. We should be able to use a loop, just properly.

Reply:
Or are you saying that doing like test_cases[test_case_count] = { blah } will make us less prone to this mistake? I can do that (and some other loops like this do), but it doesn't exactly prevent the original problem. Like, you have to already have not made the original mistake in order to be able to do test_cases[test_case_count] = { blah }.

Reply:
oh whoops, I read the diff wrong. 😅 Agreed that that isn't particularly helpful here.